spec/lib/email_alert_criteria_spec.rb

```diff
@@ -0,0 +1,135 @@
+  content_item["parent"] = %w[00f693d4-866a-4fe6-a8d6-09cd7db8980b]
+  content_item
+end
+
+def valid_content_item_no_parent
+  content_item = valid_content_item
+  content_item.delete("parent")
+  content_item
+end
+
+RSpec.describe EmailAlertCriteria do
+  include GdsApi::TestHelpers::ContentItemHelpers
+
+  describe "#would_trigger_alert?" do
+    context "when the content item has no title" do
+      let!(:content_item) { valid_content_item.except("title") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item has no base_path" do
+      let!(:content_item) { valid_content_item.except("base_path") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item has no public_updated_at" do
+      let!(:content_item) { valid_content_item.except("public_updated_at") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item is not english" do
+      let!(:content_item) { valid_content_item.merge("locale" => "cy") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item does not have a change note" do
+      let!(:content_item) do
+        ci = valid_content_item
+        ci["details"].delete("change_history")
+        ci
+      end
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item is from a blocked app" do
+      let!(:content_item) { valid_content_item.merge("publishing_app" => "travel-advice-publisher") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item is from a blocked document_type" do
+      let!(:content_item) { valid_content_item.merge("document_type" => "coming_soon") }
+
+      it "should return false" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item is otherwise valid but doesn't have a parent/tag/supertype" do
+      let!(:content_item) { valid_content_item_no_parent }
+
+      it "should return true" do
+        expect(EmailAlertCriteria.new(content_item:).would_trigger_alert?).to be false
+      end
+    end
+
+    context "when the content item is valid because it is the single instance of service_manual_service_standard" do
+      let!(:content_item) { valid_content_item_no_parent.merge("parent" => %w[00f693d4-866a-4fe6-a8d6-09cd7db8980b]) }
```

Comment:
Yeah, it's sort of weird. The problem is that you _do_ need parent in valid_content_item otherwise all the preceding tests will fail for the wrong reason. So doing it like this was a sort of way of underlining that the last sets of tests were different (they all rely on the content id being valid _except_ for the parent/links/service manual thing, then we explicitly add that in. But if that's too confusing I can just use valid_content_item for this one test.

Reply:
Right ok, because of the three-way guard clause? Makes sense.